### PR TITLE
Splunk: Feature - added additional authentication type

### PR DESCRIPTION
**Pull request by mayurah**
Please ensure your pull request (PR) adheres to the following guidelines:

APP-Tests: https://cd.splunkdev.com/phantom-apps/app-tests/-/pipelines/6603286

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- "Password" label is renamed to "Password / Token" and now supports token-based authentication in addition to existing username/password based authentication.

## What is the current behavior? (OPTIONAL)
- Current user/password based authentication does not support token-based  (Bearer) authentication.

## What is the new behavior? (OPTIONAL)
- New addition of token-based authentication lets SOAR user use token instead of password in the password field. When username is empty, password field is treated as token. This behavior is noted in `readme.html` file.

## Other information (OPTIONAL)
- https://docs.splunk.com/Documentation/Splunk/8.2.6/Security/CreateAuthTokens
- https://docs.splunk.com/Documentation/Splunk/8.2.6/Security/UseAuthTokens
- https://github.com/splunk/splunk-sdk-python#using-bearer-token

## Screenshots (if relevant)

Screen Shot 2022-05-23 at 9 15 02 PM

---
Thanks for contributing!

### CODE REVIEW

File: release_notes/unreleased.md [hunk 1]
```diff
@@ -1 +1,4 @@
 **Unreleased**
+* Added Token-based authentication workflow
```
[the remark this answers is not in the document]
Comment (btavethiya-crest) [reply release_notes/unreleased.md]: Done

File: readme.html [hunk 2]
```diff
@@ -15,6 +15,15 @@
 <html>
   <head></head>
   <body>
+
+    <p><h2>App&#39;s Token-Based Authentication Workflow</p></h2>
+    <ul>
+        <li> This app also supports API token based authentication.</li>
+        <li> Please follow the steps mentioned in this <a href="https://docs.splunk.com/Documentation/Splunk/8.2.6/Security/CreateAuthTokens" target="_blank">documentation</a> to generate an API token.</li>
+        <b>NOTE -</b>
+        If the username/password and API token both will be provided then the API token will be given preference and a Token-Based authentication workflow will be used.
```
[the remark this answers is not in the document]
Comment (btavethiya-crest) [reply readme.html]: Done

**Review by splunk-jessica — APPROVED** [review]

**Comment by mayurah** [comment]
@alexa-phantom  / @ricardo-phantom , could you pour your blessings!

🖖🏼

File: splunk_consts.py [hunk 3]
```diff
@@ -34,12 +34,18 @@
 SPLUNK_ERR_NON_NEGATIVE_INTEGER = "Please provide a valid non-negative integer value in the {param} parameter"
 SPLUNK_ERR_INVALID_PARAM = "Please provide non-zero positive integer in {param}"
 SPLUNK_ERR_MSG_UNAVAILABLE = "Error message unavailable. Please check the asset configuration and|or action parameters."
-SPLUNK_ERR_CODE_UNAVAILABLE = "Error code unavailable"
-SPLUNK_UNICODE_DAMMIT_TYPE_ERROR_MESSAGE = "Error occurred while connecting to the Splunk server. Please check the asset \
-    configuration and|or the action parameters."
-SPLUNK_EXCEPTION_ERROR_MESSAGE = "{msg}. Error Code: {error_code}. Error Message: {error_msg}"
+SPLUNK_EXCEPTION_ERROR_MESSAGE = "{msg}. {error_text}"
```
Comment (alexa-phantom) [inline splunk_consts.py:37]: Use `ERR` instead of `ERROR` for consistency

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: splunk_consts.py (hunk 3 above)
Comment (btavethiya-crest) [reply splunk_consts.py]: Done

**Review by ricardo-phantom — APPROVED** [review]

**Review by alexa-phantom — APPROVED** [review]

**Review by mayurah — COMMENTED** [review]
sgtm. what's stopping the merge?

**Pull request merged**